src/test/regress/sql/pg16.sql

```diff
@@ -650,10 +630,122 @@ RESET citus.enable_ddl_propagation;
+CREATE ROLE role7;
+GRANT role6 TO role7 WITH ADMIN OPTION, INHERIT TRUE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+
+-- connect to worker node
+\c - - - :worker_1_port
+-- run the same command
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+-- reconnect to master node
+\c - - - :master_port
+
+REVOKE role6 FROM role7;
+-- GRANT with INHERIT and SET Options
+GRANT role6 TO role7 WITH INHERIT TRUE, SET TRUE;
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+
+DROP ROLE role6, role7;
+RESET citus.log_remote_commands;
+
+-- Grant role with admin option set to true and inherit option set to false
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role9;
+CREATE ROLE role10;
+GRANT role9 TO role10 WITH ADMIN OPTION, INHERIT FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role9' ORDER BY 1, 2;
+DROP ROLE role9, role10;
+RESET citus.log_remote_commands;
+
+-- Grant role with admin option set to false and inherit option set to true
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role11;
+CREATE ROLE role12;
+GRANT role11 TO role12 WITH INHERIT FALSE, ADMIN TRUE, SET FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
+DROP ROLE role11, role12;
+RESET citus.log_remote_commands;
+
+-- Handles the inherit option
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role13;
+CREATE ROLE role14;
+GRANT role13 TO role14 WITH INHERIT FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option FROM pg_auth_members WHERE roleid::regrole::text = 'role13' ORDER BY 1, 2;
+DROP ROLE role13, role14;
+RESET citus.log_remote_commands;
+
+-- Handles the set option
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role17;
+CREATE ROLE role18;
+GRANT role17 TO role18 WITH SET TRUE;
+SET ROLE role18;
+SELECT rolname, rolinherit FROM pg_roles WHERE rolname = 'role18';
+RESET ROLE;
+DROP ROLE role17, role18;
+RESET citus.log_remote_commands;
+
+-- Handles with default inherit false
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role19;
+CREATE ROLE role20 NOINHERIT;
+GRANT role19 TO role20;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members WHERE roleid::regrole::text = 'role19' ORDER BY 1, 2;
+
+DROP ROLE role19, role20;
+RESET citus.log_remote_commands;
+
+-- show that creating role from worker node is allowed
+\c - - - :worker_1_port
+CREATE ROLE role_on_worker;
+DROP ROLE role_on_worker;
+
+\c - - - :master_port
+
```

Comment:
Here we should add a comment for the following test, and also remove the second worker node, otherwise the addition is ineffective. Something like the following:
```suggestion
-- here we test that we propagate admin, set and inherit options correctly
-- when adding a new node.

 -- First, we need to remove the node:
 SELECT 1 FROM citus_remove_node('localhost', :worker_2_port); 
```